### PR TITLE
awscurl v0.20 (new formula)

### PR DESCRIPTION
**Pull request by okigan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by jonchang** [comment]
Can you please ensure this builds, audits, and tests correctly on your local machine before opening a pull request? The contribution checklist is necessary to ensure we aren't using valuable CI time on tests that can be done locally.

File: Formula/awscurl.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    url "https://files.pythonhosted.org/packages/ad/fc/54d62fa4fc6e675678f9519e677dfc29b8964278d75333cf142892caf015/urllib3-1.25.7.tar.gz"
+    sha256 "f3c5fd51747d450d4dcf6f923c81f78f811aab8205fda64b0aba34a4e48b0745"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "Curl", shell_output("#{bin}/awscurl --help")
```
Comment (fxcoudert) [inline Formula/awscurl.rb]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: .github/workflows/check_pr.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Generate formulae.brew.sh data.
```
Comment (fxcoudert) [inline .github/workflows/check_pr.yml]: Remove this unrelated change

File: Formula/awscurl.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
+class Awscurl < Formula
+  include Language::Python::Virtualenv
+
+  desc "Curl like simplicity to access AWS resources"
+  homepage "https://github.com/okigan/awscurl"
+  url "https://github.com/okigan/awscurl/archive/263190e.tar.gz"
```
Comment (fxcoudert) [inline Formula/awscurl.rb]: Homebrew only ships stable, released versions.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/awscurl.rb (hunk 3 above)
Comment (okigan) [reply Formula/awscurl.rb]: Fixed

File: .github/workflows/check_pr.yml (hunk 2 above)
Comment (okigan) [reply .github/workflows/check_pr.yml]: Fixed

File: Formula/awscurl.rb (hunk 1 above)
Comment (okigan) [reply Formula/awscurl.rb]: This is modeled after other aws related formulas: 

https://github.com/Homebrew/homebrew-core/blob/master/Formula/awslogs.rb#L99,
https://github.com/Homebrew/homebrew-core/blob/master/Formula/awscli.rb#L50

The test was extended to cover more than the above tools.

File: Formula/awscurl.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
+class Awscurl < Formula
+  include Language::Python::Virtualenv
+
+  desc "Curl like simplicity to access AWS resources"
+  homepage "https://github.com/okigan/awscurl"
+  url "https://github.com/okigan/awscurl/archive/v0.20.tar.gz"
+  version "v0.20"
```
Comment (SMillerDev) [inline Formula/awscurl.rb]: This is not needed with the tagged version in the URL

File: Formula/awscurl.rb (hunk 1 above)
Comment (okigan) [reply Formula/awscurl.rb]: Hmm, I think I need a more specific example of testing for failure, i.e. how properly handle expected error code 1

File: Formula/awscurl.rb (hunk 4 above)
Comment (okigan) [reply Formula/awscurl.rb]: fixed

File: Formula/awscurl.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    url "https://files.pythonhosted.org/packages/ad/fc/54d62fa4fc6e675678f9519e677dfc29b8964278d75333cf142892caf015/urllib3-1.25.7.tar.gz"
+    sha256 "f3c5fd51747d450d4dcf6f923c81f78f811aab8205fda64b0aba34a4e48b0745"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    output = shell_output("#{bin}/awscurl --service s3 https://homebrew-test-none-existant-bucket.s3.amazonaws.com", "No access key is available")
```
Comment (fxcoudert) [inline Formula/awscurl.rb]:
```suggestion
    output = shell_output("#{bin}/awscurl --service s3 https://homebrew-test-none-existant-bucket.s3.amazonaws.com", 1)
```

where 1 is the expected exit code.

Comment (okigan) [reply Formula/awscurl.rb]: that still does not work -- looks like it's expecting the whole exception object there

File: Formula/awscurl.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "Curl", shell_output("#{bin}/awscurl --help")
+
+    output = shell_output("#{bin}/awscurl --service s3 https://homebrew-test-none-existant-bucket.s3.amazonaws.com", 1)
```
Comment (fxcoudert) [inline Formula/awscurl.rb]: Then check the output?
(If message is printed to stderr, you will need to include `2>&1` inside the `shell_output` command

Comment (okigan) [reply Formula/awscurl.rb]: latest update resolved the issue -- resolving this thread. What's next step for PR to be merged?

File: Formula/awscurl.rb [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "Curl", shell_output("#{bin}/awscurl --help")
+
+    shell_output("#{bin}/awscurl --service s3 https://homebrew-test-none-existant-bucket.s3.amazonaws.com", 1)
```
Comment (SMillerDev) [inline Formula/awscurl.rb]: Please check this against some expected output

**Review by fxcoudert — APPROVED** [review]
Please squash everything into a single commit

**Comment by SMillerDev** [comment]
Okay, please make the commit title `awscurl v0.20 (new formula)` as well

**Review by SMillerDev — APPROVED** [review]

**Comment by okigan** [comment]
@SMillerDev @fxcoudert question: now that PR is approved -- when would it get merged? since I cannot merge, should one be notified ?

**Comment by SMillerDev** [comment]
It's not ready to merge yet, you still need to fix the commit naming

**Comment by okigan** [comment]
@SMillerDev I see -- done.

**Comment by okigan** [comment]
@SMillerDev anything else to resolve "needs response" label and proceed to merge?
